ankihub/main/block_exam_subdecks.py

```diff
@@ -144,27 +145,136 @@ def check_block_exam_subdeck_due_dates() -> List[BlockExamSubdeckConfig]:
+        return True
+
+    except Exception as e:
+        LOGGER.error(
+            "Failed to move subdeck to main deck",
+            ankihub_deck_id=subdeck_config.ankihub_deck_id,
+            subdeck_id=subdeck_config.subdeck_id,
+            error=str(e),
+        )
+        return False
+
+
+def set_subdeck_due_date(subdeck_config: BlockExamSubdeckConfig, new_due_date: str) -> bool:
+    """Set a new due date for a block exam subdeck.
+
+    Args:
+        subdeck_config: Current subdeck configuration
+        new_due_date: New due date in YYYY-MM-DD format
+
+    Returns:
+        bool: True if successful, False otherwise
+    """
+    try:
+        updated_config = BlockExamSubdeckConfig(
+            ankihub_deck_id=subdeck_config.ankihub_deck_id, subdeck_id=subdeck_config.subdeck_id, due_date=new_due_date
+        )
+
+        config.add_block_exam_subdeck(updated_config)
+
+        LOGGER.info(
+            "Updated subdeck due date",
+            ankihub_deck_id=subdeck_config.ankihub_deck_id,
+            subdeck_id=subdeck_config.subdeck_id,
+            old_due_date=subdeck_config.due_date,
+            new_due_date=new_due_date,
+        )
+        return True
+
+    except Exception as e:
+        LOGGER.error(
+            "Failed to set subdeck due date",
+            ankihub_deck_id=subdeck_config.ankihub_deck_id,
+            subdeck_id=subdeck_config.subdeck_id,
+            new_due_date=new_due_date,
+            error=str(e),
+        )
+        return False
+
+
+def remove_block_exam_subdeck_config(subdeck_config: BlockExamSubdeckConfig) -> None:
+    """Remove a block exam subdeck configuration.
+
+    Args:
+        subdeck_config: Configuration to remove
+    """
+    config.remove_block_exam_subdeck(subdeck_config.ankihub_deck_id, subdeck_config.subdeck_id)
 
-    This is a placeholder function that will be implemented when the
-    Due Date Dialog spec is ready.
+
+def handle_expired_subdeck(subdeck_config: BlockExamSubdeckConfig) -> None:
+    """Handle an expired subdeck by showing the due date dialog.
 
     Args:
-        expired_subdecks: List of expired subdeck configurations
+        subdeck_config: Configuration of the expired subdeck
     """
-    if not expired_subdecks:
-        return
-    # TODO: Implement actual dialog triggering
-    aqt.utils.showInfo(f"{len(expired_subdecks)} block exam subdeck(s) have reached their due date.")
+    from ..gui.subdeck_due_date_dialog import SubdeckDueDateDialog
+
+    try:
+        subdeck_id = DeckId(int(subdeck_config.subdeck_id))
+        subdeck = aqt.mw.col.decks.get(subdeck_id, default=False)
+        if not subdeck:
+            LOGGER.warning("Expired subdeck not found, removing config", subdeck_id=subdeck_config.subdeck_id)
+            remove_block_exam_subdeck_config(subdeck_config)
+            return
+
+        subdeck_name = subdeck["name"].split("::")[-1]  # Get the last part after "::"
```

Comment:
:wrench: To handle nested subdecks correctly:
```suggestion
        subdeck_name = subdeck["name"].split("::", maxsplit=1)[-1] # Get name without parent deck prefix
```